src/widgets/widget_list.rs

```diff
@@ -0,0 +1,258 @@
+use crate::{
+    buffer::Buffer,
+    layout::{Corner, Rect, Size},
+    style::Style,
+    widgets::{Block, StatefulWidget, Widget},
+};
+
+use super::SizeHint;
+
+#[derive(Debug, Clone, Default)]
+pub struct WidgetListState {
+    offset: usize,
+    selected: Option<usize>,
+}
+
+impl WidgetListState {
+    pub fn selected(&self) -> Option<usize> {
+        self.selected
+    }
+
+    pub fn select(&mut self, index: Option<usize>) {
+        self.selected = index;
+        if index.is_none() {
+            self.offset = 0;
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct WidgetListItem<E: Widget + SizeHint + Clone> {
+    content: E,
+    style: Style,
+}
+
+impl<E> WidgetListItem<E>
+where
+    E: Widget + SizeHint + Clone,
+{
+    pub fn new(content: E) -> WidgetListItem<E> {
+        WidgetListItem {
+            content,
+            style: Style::default(),
+        }
+    }
+
+    pub fn style(mut self, style: Style) -> WidgetListItem<E> {
+        self.style = style;
+        self
+    }
+
+    pub fn size_hint(&self, area: &Rect) -> Size {
+        self.content.size_hint(area)
+    }
+
+    /*pub fn width(&self) -> usize {
+        self.content.width()
+    }
+    */
+}
+
+/// A widget to display several items among which one can be selected (optional)
+///
+/// # Examples
+///
+/// ```
+/// # use ratatui::widgets::{Block, Borders, List, ListItem};
+/// # use ratatui::style::{Style, Color, Modifier};
+/// let items = [ListItem::new("Item 1"), ListItem::new("Item 2"), ListItem::new("Item 3")];
+/// List::new(items)
+///     .block(Block::default().title("List").borders(Borders::ALL))
+///     .style(Style::default().fg(Color::White))
+///     .highlight_style(Style::default().add_modifier(Modifier::ITALIC))
+///     .highlight_symbol(">>");
+/// ```
+#[derive(Debug, Clone)]
+pub struct WidgetList<'a, E: Widget + SizeHint + Clone> {
+    block: Option<Block<'a>>,
+    items: Vec<WidgetListItem<E>>,
+    /// Style used as a base style for the widget
+    style: Style,
+    start_corner: Corner,
+    /// Style used to render selected item
+    highlight_style: Style,
+    /// Symbol in front of the selected item (Shift all items to the right)
+    highlight_symbol: Option<&'a str>,
```

Comment:
I was planning to remove the highlight_symbol completely, currently it is not used at all. 

But I think, I can do it, but I don't think that it will look nicer, than shifting everything (like in the list). but maybe we can make it configurable

Reply:
yes `SelectableWidget` should be way better approach. like that we give the User the freedom to create a completely  different widget for the selected item. Example: if there isn't much space the user can create one line items and the selected item shows the full information.

we still can provide a default `SelectableWidget`, which shows the arrow.

(I need first some time to explore the library. I am not sure if the `SizeHint` trait well suited for the future, I may modify it in later commits )